packages/block-library/src/social-link/index.php

```diff
@@ -45,7 +47,17 @@ function render_block_core_social_link( $attributes, $content, $block ) {
 		)
 	);
 
-	return '<li ' . $wrapper_attributes . '><a href="' . esc_url( $url ) . '" aria-label="' . esc_attr( $label ) . '" ' . $rel_target_attributes . ' class="wp-block-social-link-anchor">' . $icon . '</a></li>';
+	$link  = '<li ' . $wrapper_attributes . '>';
+	$link .= '<a href="' . esc_url( $url ) . '" aria-label="' . esc_attr( $label ) . '" title="' . esc_attr( $label ) . '" ' . $rel_target_attributes . ' class="wp-block-social-link-anchor">';
+	$link .= $icon;
+
+	if ( $show_labels ) {
+		$link .= '<span class="wp-block-social-link-label">' . esc_attr( $shown_label ) . '</span>';
```

Comment:
```suggestion
		$link .= '<span class="wp-block-social-link-label">' . esc_html( $shown_label ) . '</span>';
```

**Nit (non-blocking:** These two functions are basically the same, but I think `esc_html` makes more sense in this context.